src/gan/discriminator/model.py

```diff
@@ -60,13 +63,19 @@ def __init__(self, args, vocab): #vocab_size, in_dim, mem_dim, sparsity, freeze)
                                 sparse=self.sparsity)
         if self.freeze:
             self.emb.weight.requires_grad = False
-        self.childsumtreelstm = ChildSumTreeLSTM(self.in_dim, self.mem_dim)
-        self.confidence = nn.Linear(self.mem_dim, 1)
+        #self.childsumtreelstm = ChildSumTreeLSTM(self.in_dim, self.mem_dim)
+        
+        self.discriminator_cnn = nn.Sequential(
+                    nn.Conv1d(in_channels = self.in_dim, out_channels = 1,  
+                        kernel_size = 1, stride = 1, padding = 1),
+                    nn.ReLU(inplace = True),
+                    nn.BatchNorm1d(1)) 
+        
```

Comment:
Diana,

This is a good start, but I think we will want to make the CNN implementation somewhat more complex to capture more about the longer range dependencies in the inputs. Checkout the `cnn_classifier()` class in our verbatim coding repo - it's a pytorch implementation of a classic CNN text classifier system that should work quite well (https://arxiv.org/abs/1408.5882 for the source paper).

Reply:
Updated files on CNN_Discriminator branch.